File: Cocoa Programming/Cocoa Pr L67 PresentingVC/Cocoa Pr L67 PresentingVC/InitialViewController.m
//
//  InitialViewController.m
//  Cocoa Pr L67 PresentingVC
//
//  Created by Michele Galvagno on 08/07/22.
//

#import "InitialViewController.h"
#import "ModalViewController.h"
#import "ModalAnimator.h"

@interface InitialViewController ()

@end

@implementation InitialViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [[self.view layer] setBackgroundColor:NSColor.systemRedColor.CGColor];
}

- (IBAction)presentModalVC:(NSButton *)sender {
    NSLog(@"Present button pressed");
    ModalAnimator *animator = [[ModalAnimator alloc] init];
    ModalViewController *modalVC = [[ModalViewController alloc] init];
    [self presentViewController:modalVC animator:animator];
}

@end

File: Cocoa Programming/Cocoa Pr L67 Swift/Cocoa Pr L67 Swift/InitialViewController.swift
//
//  InitialViewController.swift
//  Cocoa Pr L67 Swift
//
//  Created by Michele Galvagno on 07/07/22.
//

import Cocoa

class InitialViewController: NSViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        view.layer?.backgroundColor = NSColor.systemRed.cgColor
    }
    
    @IBAction func presentModalVC(_ sender: NSButton) {
        let modalVC = ModalViewController()
        present(modalVC, animator: ModalAnimator())
    }
    
}
